src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -1,751 +1,5624 @@
+
+            nint tempLinear = TensorPrimitives.Product(newLengths);
+            if (tempLinear != tensor.FlattenedLength)
+                ThrowHelper.ThrowArgument_InvalidReshapeDimensions();
+
+            nint[] strides;
+
+            // If all our strides are 0 we can reshape however we like and keep all new strides at 0
+            if (!tensor.Strides.ContainsAnyExcept(0))
+            {
+                strides = new nint[newLengths.Length];
+            }
+            // If we contain a 0 stride we can only add dimensions of length 1.
+            else if (tensor.Strides.Contains(0))
+            {
+                List<nint> origStrides = new List<nint>(tensor.Strides.ToArray());
+                int lengthOffset = 0;
+                for (int i = 0; i < newLengths.Length; i++)
+                {
+                    if (lengthOffset < tensor.Rank && newLengths[i] == tensor.Lengths[lengthOffset])
+                        lengthOffset++;
+                    else if (newLengths[i] == 1)
+                    {
+                        if (lengthOffset == tensor.Rank)
+                            origStrides.Add(tensor.Strides[lengthOffset - 1]);
+                        else
+                            origStrides.Insert(i, tensor.Strides[i] * tensor.Lengths[i]);
+                    }
+                    else
+                        ThrowHelper.ThrowArgument_InvalidReshapeDimensions();
+                }
+                strides = origStrides.ToArray();
+            }
+            else
+                strides = [];
+
+            ReadOnlyTensorSpan<T> output = new ReadOnlyTensorSpan<T>(ref tensor._reference, tensor._shape.LinearLength, newLengths, strides);
+            return output;
+        }
+        #endregion
+
+        #region Resize
+        /// <summary>
+        /// Creates a new <see cref="Tensor{T}"/>, allocates new memory, and copies the data from <paramref name="tensor"/>. If the final shape is smaller all data after
+        /// that point is ignored.
+        /// </summary>
+        /// <param name="tensor">Input <see cref="Tensor{T}"/>.</param>
+        /// <param name="lengths"><see cref="ReadOnlySpan{T}"/> of the desired new shape.</param>
+        public static Tensor<T> Resize<T>(Tensor<T> tensor, ReadOnlySpan<nint> lengths)
+        {
+            nint newSize = TensorPrimitives.Product(lengths);
+            T[] values = tensor.IsPinned ? GC.AllocateArray<T>((int)newSize) : (new T[newSize]);
+            Tensor<T> output = Tensor.Create(values, 0, lengths, []);
+            ReadOnlySpan<T> span = MemoryMarshal.CreateSpan(ref Unsafe.Add(ref tensor.AsTensorSpan()._reference, tensor._start), (int)tensor._values.Length - tensor._start);
+            Span<T> ospan = MemoryMarshal.CreateSpan(ref output.AsTensorSpan()._reference, (int)output.FlattenedLength);
+            if (newSize >= span.Length)
+                span.CopyTo(ospan);
+            else
+                span.Slice(0, ospan.Length).CopyTo(ospan);
```

Comment:
Can this replaced by calling `ResizeTo<T>(scoped in Tensor<T> tensor, in TensorSpan<T> destination)`?

Reply:
Yes, fixed locally for followup pr.